pkg/scheduler/api/resource_info.go

```diff
@@ -208,30 +245,23 @@ func (r *Resource) FitDelta(rr *Resource) *Resource {
 		r.Memory -= rr.Memory + minResource
 	}
 
-	for rrName, rrQuant := range rr.ScalarResources {
-		if r.ScalarResources == nil {
-			r.ScalarResources = map[v1.ResourceName]float64{}
-		}
+	if r.ScalarResources == nil {
+		r.ScalarResources = make(map[v1.ResourceName]float64)
+	}
 
-		if rrQuant > 0 {
+	for rrName, rrQuant := range rr.ScalarResources {
+		_, ok := r.ScalarResources[rrName]
+		if ok && rrQuant > 0 {
```

Comment:
if r does not have `rrName` also need do `r.ScalarResources[rrName] -= rrQuant + minResource`

Reply:
> if r does not have `rrName` also need do `r.ScalarResources[rrName] -= rrQuant + minResource`

If that, we should initially set `r.ScalarResources[rrName] == 0`